### PR TITLE
fix(compiler): add more specific matcher for hydrate never block

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #58358

## What is the new behavior?

No additional characters are allowed in the "hydrate never" block.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by devversion — APPROVED** [review]

**Review by thePunderWoman — APPROVED** [review]
LGTM, never, and thank you :egg:

**Comment by thePunderWoman** [comment]
Caretaker note: the broken adev tests are broken at head and unrelated to this change. This is safe to merge.

**Review by thePunderWoman — CHANGES_REQUESTED** [review]

File: packages/compiler/src/render3/r3_deferred_blocks.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const HYDRATE_WHEN_PATTERN = /^hydrate\s+when\s/;
 const HYDRATE_ON_PATTERN = /^hydrate\s+on\s/;
 
 /** Pattern to identify a `hydrate never` trigger. */
-const HYDRATE_NEVER_PATTERN = /^hydrate\s+never\s*/;
+const HYDRATE_NEVER_PATTERN = /^hydrate\s+never$/;
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply packages/compiler/src/render3/r3_deferred_blocks.ts]: Good catch. Updated the expression to allow only additional spaces and added some additional tests

**Review by thePunderWoman — APPROVED** [review]
Thanks for the additional tests!

**Comment by thePunderWoman** [comment]
@brandonroberts Can you rebase on main? That should fix the broken adev tests.

**Comment by brandonroberts** [comment]
@thePunderWoman yep, done

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 992410e928f15308edb37a165a6b59d2c94c30ba.

The changes were merged into the following branches: main, 19.0.x

**Pull request closed without merging**